cmd/generate/BuildImageDockerfile.template

```diff
@@ -1,16 +1,13 @@
 # DO NOT EDIT! Generated Dockerfile.
 
+FROM registry.ci.openshift.org/ocp/4.17:tools as tools
```

Comment:
I'm wondering if there's a way to not hardcode `4.17` here.

Reply:
I didn't see any "latest" (but also don't know a way to browse the registry). We could also try to get it from the builder image name, but that's unsafe I guess.

Reply:
Just an idea. Maybe put another template next to this one? https://github.com/openshift-knative/hack/blob/main/cmd/generate/generate.go#L88
I know we already have a few arguments but having it sid-by-side with the golang builder could ensure that we will be updating it at the same time.

Reply:
But we can also leave it as it is. This will be changing rarely.

Reply:
I use a constant now for the image and added a comment with some details & "update recommendations" for it

Reply:
**NOTE:** It's important that the tools image and the builder image use same version of glibc.

Reply:
or at least the builder image has newer glibc than tool image.